src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java

```diff
@@ -226,9 +227,34 @@ public void validate() {
     private void initGoogleCredentials() {
         synchronized (this) {
             if (null == googleCredentials) {
-                UserCredentials.Builder credentialsBuilder = UserCredentials.newBuilder()
-                        .setClientId(getClientId())
-                        .setClientSecret(SecurityUtil.decrypt(getClientSecret()));
+                UserCredentials.Builder credentialsBuilder = UserCredentials.newBuilder();
+
+                Proxy.Type proxyType =
+                        (System.getProperty("http.proxyHost") != null && System.getProperty("http.proxyPort") != null)
+                                || (System.getProperty("https.proxyHost") != null
+                                && System.getProperty("https.proxyPort") != null)
+                                ? Proxy.Type.HTTP
+                                : System.getProperty("socksProxyHost") != null
+                                        && System.getProperty("socksProxyPort") != null
+                                        ? Proxy.Type.SOCKS
+                                        : Proxy.Type.DIRECT;
+                
+                if (Proxy.Type.HTTP == proxyType) {
+                    HttpClientBuilder clientBuilder = HttpClientBuilder.create();
+                    clientBuilder.useSystemProperties();
```

Comment:
AFAIK `useSystemProperties()` will configure this builder to take the proxy configuration (among other things) from system properties, hence I think all the statement below should be safely removed.

https://www.demo2s.com/java/apache-httpclient-httpclientbuilder-tutorial-with-examples.html

Reply:
already tried without the further code, but still does not work without explicit proxy configuration. Going to investigate better.

Reply:
I see.

Can I suggest anyway to read the system properties just one, to improve readability? Here and in the other modified code.

Reply:
Sure much better! Going to do this

Reply:
Confirm from DEBUG logging on `org.apache.http.impl.conn` version `4.5.14` that  system properties are not taken into account. That's really strange

Reply:
Ok, so let's leave the code with explicit settings as it is